veps/sig-compute/decentralized-migration/decentralized-migration-vep.md

```diff
@@ -0,0 +1,308 @@
+ - Create a synchronization flow to allow the synchronization of two `VirtualMachineInstance` statuses.
+ - Update the `VirtualMachineInstance` migrationState to support those separate entities and to allow the virt-controller, and virt-handler to identify which entity they are processing.
+
+## Non Goals
+
+- Orchestration of the building blocks provided by this proposal. A higher level orchestrator is required to make usage easier for end users.
+- Ensuring connectivity between clusters if live migrating between clusters.
+- Creating and managing required resources needed to migrate a `Virtual Machine` to the target. It is assumed all the required resources will be available at the target.
+
+## Definition of Users
+
+- Cluster Admins
+- Admins managing several clusters
+
+## User Stories
+
+- As a KubeVirt admin I would like to evacuate my cluster so I can do disruptive maintance on this cluster (ex. hardware refresh) without disrupting the workloads.
+- As a KubeVirt admin maintaining several kubernetes clusters, I would like to rebalance the virtual machines running on my clusters to even out the load of my clusters. without disrupting the workloads.
+- As a KubeVirt namespace admin, I would like to move a virtual machine from one namespace to another without disrupting the workloads.
+
+## Repos
+
+[KubeVirt](https://github.com/kubevirt/kubevirt)
+
+## Design
+
+#### Current state
+To start a live migration one creates a `VirtualMachineInstanceMigration` resource in the cluster, which references the `VirtualMachineInstance` you want to migrate. This then creates a _receiver_ pod in the same namespace, and starts the migration process.
+
+The actual migration channel is proxied through the `virt-handler` pods that are running on the source and target nodes. Any migration state information is passed through the `VirtualMachineInstance` migrationStatus. This is because `virt-handler` pods can only read/modify `VirtualMachineInstances`.
+
+The reason virt-launcher pods do not communicate directly with each other, is that it is not guaranteed that virt-launchers will have network connectivity. The virt-handler pods also have access to a migration network if one is defined.
+
+The migration channel in figure 1 is either the pod network, or a specially configured migration network.
+
+Figure 1 describes the current interaction between various components:
+![figure 1](migration_current_state.png)
+
+#### Proposed state
+To start a live migration one now has to create two `VirtualMachineInstanceMigration` resources. A `target` resource, and a `source` resource. Additionally in order to migrate a special `VirtualMachine` resource has to be created on the target which is copy of the source `VirtualMachine` but it has a special `runStrategy` indicating it is the receiving `VirtualMachine`. The virt-controller will create a special receiving `VirtualMachineInstance` that can be referenced by the `target` `VirtualMachineInstanceMigration`.
```

Comment:
I'm not a big fan of adding this to the runStrategy. Once, the migration has completed then you need to switch the runStrategy to something else, who is going to to this?

Reply:
The virt-controller, and we even know what the runStrategy needs to be, because we can take it from the original VMI.

Reply:
The spec should the part that the users modify. I feel that this information of the state of the VMI should be in the status not in the spec